private/signer/v4/v4.go

```diff
@@ -326,7 +326,7 @@ func (v4 *signer) buildCanonicalHeaders(r rule, header http.Header) {
 			headerValues[i] = "host:" + v4.Request.URL.Host
 		} else {
 			headerValues[i] = k + ":" +
-				strings.Join(v4.Request.Header[http.CanonicalHeaderKey(k)], ",")
+				strings.Join(v4.signedHeaderVals[k], ",")
```

Comment:
Could you elaborate? I am sure that I am just simply not understanding it.

The problem I hit is that lowercasing the key, then calling `http.CanonicalHeaderKey` on the result does not necessarily get you back to the original key in order to retrieve the header value.  The value, however, has already been stored using the lowercased key in `signedHeaderVals` so it seemed that we would be able to simply retrieve it without needing to make the route trip of obtaining the original key.

I am glad to look at an alternative solution.